tests/IBTK/snapshot_cache_01.cpp

```diff
@@ -0,0 +1,251 @@
+        pout << "Testing with side variable\n";
+        test<SideVariable<NDIM, double> >("side",
+                                          old_patch_hierarchy,
+                                          new_patch_hierarchy,
+                                          fcn,
+                                          time_pts,
+                                          interp_pts,
+                                          t_start,
+                                          t_end,
+                                          "CONSERVATIVE_LINEAR_REFINE",
+                                          -1);
+        pout << "Testing with edge variable\n";
+        test<EdgeVariable<NDIM, double> >("edge",
+                                          old_patch_hierarchy,
+                                          new_patch_hierarchy,
+                                          fcn,
+                                          time_pts,
+                                          interp_pts,
+                                          t_start,
+                                          t_end,
+                                          "CONSERVATIVE_LINEAR_REFINE",
+                                          -1);
+        pout << "Testing with face variable\n";
+        test<FaceVariable<NDIM, double> >("face",
+                                          old_patch_hierarchy,
+                                          new_patch_hierarchy,
+                                          fcn,
+                                          time_pts,
+                                          interp_pts,
+                                          t_start,
+                                          t_end,
+                                          "CONSERVATIVE_LINEAR_REFINE",
+                                          -1);
+    }
+} // main()
+
+template <class VariableType>
+void
+test(const std::string& test_name,
+     Pointer<PatchHierarchy<NDIM> > cur_hierarchy,
+     Pointer<PatchHierarchy<NDIM> > new_hierarchy,
+     Pointer<muParserCartGridFunction> fcn,
+     const std::set<double>& time_pts,
+     const std::set<double>& interp_pts,
+     const double t_start,
+     const double t_end,
+     const std::string& refine_type,
+     const int wgt_idx)
+{
+    // Actually do the test.
+    // Create the index that we fill with data
+    Pointer<VariableType> var = new VariableType(test_name);
+    auto var_db = VariableDatabase<NDIM>::getDatabase();
+    int var_idx = var_db->registerVariableAndContext(var, var_db->getContext("ctx"), 1 /*ghosts*/);
+    int err_idx = var_db->registerVariableAndContext(var, var_db->getContext("err"));
+
+    // Create a hierarchy interpolator.
+    SnapshotCache<VariableType> snapshot_cache(test_name + "::SnapshotCache");
+    auto hier_math_ops = HierarchyDataOpsManager<NDIM>::getManager();
+    Pointer<HierarchyDataOpsReal<NDIM, double> > hier_data_ops = hier_math_ops->getOperationsDouble(var, new_hierarchy);
```

Comment:
```suggestion
    Pointer<HierarchyDataOpsReal<NDIM, double> > hier_data_ops = hier_math_ops->getOperationsDouble(var, new_hierarchy, true);
```
We always want unique math ops to avoid leaks.